src/sync/polling/updaters/splitChangesUpdater.ts

```diff
@@ -31,7 +31,10 @@ function checkAllSegmentsExist(segments: ISegmentsCacheBase): Promise<boolean> {
  */
 export function parseSegments(ruleEntity: ISplit | IRBSegment, matcherType: typeof IN_SEGMENT | typeof IN_RULE_BASED_SEGMENT = IN_SEGMENT): Set<string> {
   const { conditions = [], excluded } = ruleEntity as IRBSegment;
-  const segments = new Set<string>(excluded && excluded.segments);
+
+  const segments = new Set<string>(
+    excluded?.segments?.map(segment => segment.name) || []
```

Comment:
```suggestion
    excluded && excluded.segments ? excluded.segments.map(segment => segment.name) : []
```

Reply:
To ensure compatiblity across environments without support for optional chaining and avoid transpiling something new, I suggest we take this simpler approach with a ternary operator and truthiness casting.